### ISSUE TITLE
Feature request: quick calculate specific hash menu option (in ListView rightclick)

### ISSUE THREAD

**Issue by Nakashio** [OV2/RapidCRC-Unicode#59]
An improvement suggestion for this great program:
Imagine that you encounter a hash of an unusual algorithm provided on a webpage for a downloaded file that's you don't normally use and don't have configured in RapidCRC Unicode. The current procedure for the user to verify the file in this scenario AFAIK is to enter the program settings, turn on checking for the new algorithm, (re)check the file, rightclick it to copy hash (or turn on viewing the algorithm in ListView too), then revert the changes to the settings you made. Quite inefficient.

I'm proposing the rightclick context menu in ListView to directly allow calculating with non-pre-set algorithms:  instead having a useless menu item like a grayed out "Copy ED2K to clipboard", it would be swapped with one like "Calculate ED2K hash" instead. When you press it, it will do so (and ion the next rightclick, you'll now get the copy to clipboard item instead), and it should probably also automatically add the algorithm to be visible in ListView, temporarily, until the program is closed.

**Comment by vatterspun** [comment]
So if I'm reading this right:

**Current behavior**: To get all the available hashes, you must have them selected in the "when not checking" section of the options menu.  

Requested behavior: Right click on file and select "copy the ED2K link to clipboard" and it basically just runs a hash to get that pasted to the clipboard.

**Why not just enable in settings**: Running ALL the hashes every time I check a file is inefficient, but the right-click menu should enable easy access to lesser-used algorithms.

**Comment by Nakashio** [comment]
Yes. You don't want to calculate all the hashes on a regular basis, so currently you have to temporarily change the settings, calculate and then change them back, so a right-click shortcut to use instead would be much quicker.

**Comment by OV2** [comment]
I will take a look at it when I find some time to work on rcrc again. The main problem with this is that rcrc is not built to calculate the hash on individual files, only on the whole job. So it might have to recalculate the new hash for the whole job(s) from which you selected a file(s), just like it does right now when you want to create a specific hash file and that type has not been calculated for the selected files.

**Comment by vatterspun** [comment]
> not built to calculate the hash on individual files, only on the whole job

Weird.  Well, if RCRC just does them all then that's not really more efficient.  Sigh ... here I was thinking CRC32 was crazy fast and I was saving CPU by only running one algorithm and not all the others.

Anyway I guess @Nakashio 's idea is still valid if you want to avoid a cluttered UI.
